module/os/linux/zfs/abd.c

```diff
@@ -557,23 +581,30 @@ abd_free_pages(abd_t *abd)
 		}
 	}
 
-	vmem_free(ABD_SCATTER(abd).abd_sgl, n * sizeof (struct scatterlist));
+	abd_free_sg_table(abd);
 }
 
 #endif /* _KERNEL */
 
 void
 abd_init(void)
 {
-	int i;
-
 	abd_cache = kmem_cache_create("abd_t", sizeof (abd_t),
 	    0, NULL, NULL, NULL, NULL, NULL, 0);
 
+	abd_zero_buf = zio_buf_alloc(SPA_MAXBLOCKSIZE);
+	(void) memset(abd_zero_buf, 0, SPA_MAXBLOCKSIZE);
+#ifdef ZFS_IS_GPL_COMPATIBLE
+#ifdef _KERNEL
```

Comment:
We'll probably want to drop this in the final version, but it is useful for debugging.  This could also be rewritten as:

```
#if defined(ZFS_IS_GPL_COMPATIBLE) && defined(_KERNEL)
...
#endif
```